dart/lib/src/sentry_stack_trace_factory.dart

```diff
@@ -0,0 +1,121 @@
+
+  /// A list of string prefixes of module names that belong to the app. This option takes precedence
+  /// over inAppExcludes.
+  List<String> _inAppIncludes;
+
+  SentryStackTraceFactory(SentryOptions options) {
+    if (options == null) {
+      throw ArgumentError('SentryOptions is required.');
+    }
+
+    _inAppExcludes = options.inAppExcludes;
+    _inAppIncludes = options.inAppIncludes;
+  }
+
+  /// returns the [SentryStackFrame] list from a stackTrace ([StackTrace] or [String])
+  List<SentryStackFrame> getStackFrames(dynamic stackTrace) {
+    if (stackTrace == null) return null;
+
+    final chain = stackTrace is StackTrace
+        ? Chain.forTrace(stackTrace)
+        : Chain.parse(stackTrace as String);
+
+    final frames = <SentryStackFrame>[];
+    for (var t = 0; t < chain.traces.length; t += 1) {
+      final encodedFrames =
+          chain.traces[t].frames.map((f) => encodeStackTraceFrame(f));
+
+      frames.addAll(encodedFrames);
+
+      if (t < chain.traces.length - 1) {
+        frames.add(SentryStackFrame.asynchronousGapFrameJson);
+      }
+    }
+
+    return frames.reversed.toList();
+  }
+
+  /// converts [Frame] to [SentryStackFrame]
+  @visibleForTesting
+  SentryStackFrame encodeStackTraceFrame(Frame frame) {
+    final fileName =
+        frame.uri.pathSegments.isNotEmpty ? frame.uri.pathSegments.last : null;
+
+    var sentryStackFrame = SentryStackFrame(
+      absPath: '$eventOrigin${_absolutePathForCrashReport(frame)}',
+      function: frame.member,
+      // https://docs.sentry.io/development/sdk-dev/features/#in-app-frames
+      inApp: isInApp(frame),
+      fileName: fileName,
+      package: frame.package,
+    );
+
+    if (frame.line != null && frame.line >= 0) {
+      sentryStackFrame = sentryStackFrame.copyWith(lineNo: frame.line);
+    }
+
+    if (frame.column != null && frame.column >= 0) {
+      sentryStackFrame = sentryStackFrame.copyWith(colNo: frame.column);
+    }
+
+    return sentryStackFrame;
+  }
+
+  /// A stack frame's code path may be one of "file:", "dart:" and "package:".
+  ///
+  /// Absolute file paths may contain personally identifiable information, and
+  /// therefore are stripped to only send the base file name. For example,
+  /// "/foo/bar/baz.dart" is reported as "baz.dart".
+  ///
+  /// "dart:" and "package:" imports are always relative and are OK to send in
+  /// full.
+  String _absolutePathForCrashReport(Frame frame) {
+    if (frame.uri.scheme != 'dart' && frame.uri.scheme != 'package') {
+      return frame.uri.pathSegments.last;
+    }
+
+    return '${frame.uri}';
+  }
+
+  /// whether this frame comes from the app and not from Dart core or 3rd party librairies
+  bool isInApp(Frame frame) {
+    final scheme = frame.uri.scheme;
+
+    if (scheme == null || scheme.isEmpty) {
+      return true;
+    }
+
+    if (_inAppIncludes != null) {
+      for (final include in _inAppIncludes) {
+        if (frame.package.startsWith(include)) {
```

Comment:
`package` can be null I guess
https://pub.dev/documentation/stack_trace/latest/stack_trace/Frame/package.html

Reply:
✅good catch !